### DISCUSSION TITLE
Docker doesn't install `cli-plugins` correctly.

### DISCUSSION THREAD

**Discussion by jmuchovej** [Homebrew/discussions#5040] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: b3751bca8c4a3c76107d5aa3b75b81db93cf4bb6
Last commit: 2 days ago
Core tap JSON: 10 Jan 02:43 UTC
Core cask tap JSON: 10 Jan 03:53 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BROWSER: open
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.aZ7ISXbJym/org.xquartz:0
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 20
HOMEBREW_NO_ANALYTICS: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 20-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

When installing the `docker` cask, `cli-plugins` are symlink'd into `~/.docker/cli-plugins`. While this is supposed to be the intended location for plugins to be installed, an installation like this is left broken and semi-functional (that is, plugins like `buildx`, `compose`, etc. don't work).

Per this issue https://github.com/docker/for-mac/issues/6928#issuecomment-1728442544, I was able to get Docker (installed via `brew`) working and usable, by linking `/usr/local/lib/docker/cli-plugins -> /Applications/Docker.app/Contents/Resources/cli-plugins`.

I'm opening this here because I'm not certain if this is widespread _and_ if it should be a patch made by `brew` until upstream resolves it.

**For clarity**, Docker does not fail to install, _rather it fails to install with full functionality of Docker Desktop._

### Relevant casks

docker

**Comment by mklnz** [comment]
Having this issue as well, maybe .docker folder needs to exist beforehand? On a different Mac I don't recall having this issue

**Reply by jmuchovej** [reply]
I think my issue, actually, was that I set `$DOCKER_CONFIG` to `~/.config/docker`, but Homebrew didn't install the `cli-plugins` there, instead it installed to `~/.docker`. I ended up switching to `colima` though since Docker Desktop was turning out to be more hassle than I wanted to deal with. 😅
